### PR TITLE
Allow aliases to appear in top-level help

### PR DESCRIPTION
**Pull request by mterrel**
A couple of background notes on the implementation here:

- The hope here is to eventually get a PR submitted upstream with similar functionality.
- We don't want to change the default behavior of aliases & help.
- The @oclif repos are very tightly coupled when it comes to the Command type, where aliases are configured.
- The entirety of configuration of aliases is a static string array of names on the Command class, so there's no good place to put additional alias config without also changing Command (which would pretty much require us to fork all remaining @oclif repos, just to maintain that type difference).

Due to the above considerations, I chose to configure the non-default alias behavior in the "oclif" section of package.json and to require explicit opt-in for every alias you want to show up in the top level help.

### CODE REVIEW

File: test/root.test.ts [hunk 1]
```diff
@@ -6,7 +6,8 @@ g.columns = 80
 import Help from '../src'
 
 const VERSION = require('../package.json').version
-const UA = `@oclif/plugin-help/${VERSION} ${process.platform}-${process.arch} node-${process.version}`
+const NAME = require('../package.json').name
+const UA = `${NAME}/${VERSION} ${process.platform}-${process.arch} node-${process.version}`
```
Comment (mvachhar) [inline test/root.test.ts:10]: Should this commit be pulled out into a separate PR to merge upstream?

**Review by mvachhar — CHANGES_REQUESTED** [review]
How hard is it to add a test case?  Does this change make all aliases show up or is it configured alias by alias, without a test case it is a bit hard to tell.

File: test/root.test.ts (hunk 1 above)
Comment (mterrel) [reply test/root.test.ts:10]: Yes, there should be 2 separate PRs upstream. And there really needs to be a little infrastructure created in order to really test this feature. We would need to be able to create a little test project that uses plugin-help instead of using plugin-help to test itself, as is currently done.
